server/data/prisonRegisterApiClient.test.ts

```diff
@@ -0,0 +1,67 @@
+import nock from 'nock'
+import config from '../config'
+import PrisonRegisterService from '../services/prisonRegisterService'
+import HmppsAuthClient from './hmppsAuthClient'
+import type { PrisonDto } from '../@types/prisonRegisterTypes'
+
+jest.mock('./hmppsAuthClient')
+
+const hmppsAuthClient = new HmppsAuthClient(null) as jest.Mocked<HmppsAuthClient>
+
+const prisonRegisterService = new PrisonRegisterService(hmppsAuthClient)
+
+const prisonDto = {
+  prisonId: 'LEI',
+  prisonName: 'Leeds HMP',
+  active: true,
+} as PrisonDto
+
+describe('Prison register client', () => {
+  let fakeApi: nock.Scope
+  const username = 'joebloggs'
+  const prisonCode = 'LEI'
+
+  beforeEach(() => {
+    config.apis.prisonRegisterApi.url = 'http://localhost:8100'
+    fakeApi = nock(config.apis.prisonRegisterApi.url)
+  })
+
+  afterEach(() => {
+    nock.cleanAll()
+  })
+
+  describe('Get prison description', () => {
+    it('Valid request', async () => {
+      hmppsAuthClient.getSystemClientToken.mockResolvedValue('a token')
+      fakeApi.get(`/prisons/id/${prisonCode}`).reply(200, prisonDto)
+      const data = await prisonRegisterService.getPrisonDescription(username, prisonCode)
+      expect(data).toEqual(prisonDto)
+      expect(nock.isDone()).toBe(true)
+      expect(hmppsAuthClient.getSystemClientToken).toBeCalled()
+    })
+
+    it('No client token', async () => {
+      hmppsAuthClient.getSystemClientToken.mockResolvedValue('')
+      fakeApi.get(`/prisons/id/${prisonCode}`).reply(401)
```

Comment:
I thought it was only the `/secure` endpoints that required a token?

Reply:
Yes, you're right. First time using prison register for me. I'll sort that now.

Reply:
Looking at this, and at other service implementations, some have amended the rest client to make the token nullable, and then conditionally made the call to superagent either with of without the authorization header if no token is provided. I'm just thinking that leaving this with the token always in place is probably less harmful and simpler? As an insecure endpoint it just won't check the auth header even when provided?

Reply:
Hmmm I think you're right, though it feels a bit dirty.